aospy/utils/times.py

```diff
@@ -559,17 +559,16 @@ def ensure_time_as_dim(ds):
     Returns
     -------
     Dataset
+
     """
-    if TIME_STR not in ds.dims:
-        time = convert_scalar_to_indexable_coord(ds[TIME_STR])
-        ds = ds.set_coords(TIME_STR)
-        for name in ds.variables:
-            if ((name not in GRID_ATTRS_NO_TIMES) and
-               (name != TIME_STR)):
-                da = ds[name]
-                da, _ = xr.broadcast(da, time)
-                da[TIME_STR] = time
-                ds[name] = da
+    time_indexed_vars = set(ds.data_vars).union({TIME_WEIGHTS_STR,
+                                                 TIME_BOUNDS_STR})
+    time_indexed_vars = time_indexed_vars.intersection(ds.variables)
+    for name in time_indexed_vars:
+        if TIME_STR not in ds[name].indexes:
+            da = ds[name].expand_dims(TIME_STR)
+            da[TIME_STR] = convert_scalar_to_indexable_coord(ds[TIME_STR])
```

Comment:
I'm slightly partial to a solution that eliminates `convert_scalar_to_indexable_coord` for the simple reason that it is something that Xarray supports now (so we shouldn't need to roll our own solution)